### PR TITLE
Update the Contribution Guide

### PR DESCRIPTION
**Pull request by svekars**

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: docs/source/community/contribution_guide.rst [hunk 1]
```diff
@@ -2,18 +2,19 @@ PyTorch Contribution Guide
 ==========================
 
 PyTorch is a GPU-accelerated Python tensor computation package for
-building deep neural networks built on tape-based autograd systems.
+building deep neural networks based on tape-based autograd systems.
```
Comment (albanD) [inline docs/source/community/contribution_guide.rst:5]: based twice? maybe "using a"

File: docs/source/community/contribution_guide.rst [hunk 2]
```diff
@@ -2,18 +2,19 @@ PyTorch Contribution Guide
...
+--------------------
 
 The PyTorch organization is governed by :doc:`PyTorch
-Governance <governance>`.
+Governance <governance>` and the technical guide to contributing
+can be found in `CONTRIBUTING.md <https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md>`_.
 
 The PyTorch development process involves a healthy amount of open
-discussions between the core development team and the community.
+discussions between the core development team and community.
```
Comment (albanD) [inline docs/source/community/contribution_guide.rst:15]: shouldn't we keep the "the"?

File: docs/source/community/contribution_guide.rst [hunk 3]
```diff
@@ -30,7 +31,8 @@ here is the basic process.
       We also maintain some labels for issues that are likely to be
       good for new people, e.g., **bootcamp** and **1hr**, although
       these labels are less well maintained.
-   -  Join us on Slack and let us know you're interested in getting to
+   -  Join us on `dev discuss <https://dev-discuss.pytorch.org/>`_
```
Comment (albanD) [inline docs/source/community/contribution_guide.rst:34]: Adding the dev forum is a great addition here. But I think slack is still a better place for people to say they're looking for a task to pick up.
The forum is more of a place to ask question or read about design discussions between core devs.

File: docs/source/community/contribution_guide.rst [hunk 4]
```diff
@@ -287,7 +294,7 @@ Frequently asked questions
    tasks or pull requests with your environment details is helpful and
    appreciated.
 -  **CI tests failed, what does it mean?** Maybe you need to merge with
-   master or rebase with latest changes. Pushing your changes should
+   master or rebase with the latest. Pushing your changes should
```
Comment (albanD) [inline docs/source/community/contribution_guide.rst:297]: I think a better phrasing would be: "Maybe your PR is based off a broken master, you can try to rebase your change on top of latest master. You can also see the current status of master's CI at https://hud.pytorch.org/"

**Review by albanD — COMMENTED** [review]
Great improvements! Thanks!

File: docs/source/community/contribution_guide.rst (hunk 1 above)
Comment (orionr) [reply docs/source/community/contribution_guide.rst:5]: I wrote the original copy, so looks good! @svekars can you change?

File: docs/source/community/contribution_guide.rst (hunk 2 above)
Comment (orionr) [reply docs/source/community/contribution_guide.rst:15]: I'm okay either way

File: docs/source/community/contribution_guide.rst (hunk 3 above)
Comment (orionr) [reply docs/source/community/contribution_guide.rst:34]: Unfortunately slack is restricted right now (people need to be manually added) and the idea was to promote the option that was available to everyone. Thoughts?

Comment (orionr) [reply docs/source/community/contribution_guide.rst:34]: We can iterate on this later too

File: docs/source/community/contribution_guide.rst (hunk 4 above)
Comment (orionr) [reply docs/source/community/contribution_guide.rst:297]: Yup - love it. @svekars can you make the change?

**Review by orionr — COMMENTED** [review]
Just one comment about the marketing email. Thanks!

File: docs/source/community/contribution_guide.rst (hunk 1 above)
Comment (svekars) [reply docs/source/community/contribution_guide.rst]: Addressed.

File: docs/source/community/contribution_guide.rst (hunk 2 above)
Comment (svekars) [reply docs/source/community/contribution_guide.rst]: Added `the` back.

File: docs/source/community/contribution_guide.rst (hunk 4 above)
Comment (svekars) [reply docs/source/community/contribution_guide.rst]: Yup, done.

File: docs/source/community/contribution_guide.rst [hunk 5]
```diff
@@ -193,17 +200,17 @@ Promoting PyTorch
 Your use of PyTorch in your projects, research papers, write ups, blogs,
 or general discussions around the internet helps to raise awareness for
 PyTorch and our growing community. Please reach out to
-`pytorch-marketing@fb.com <http://mailto:pytorch-marketing@fb.com/>`__
+`marketing@pytorch.org <mailto:marketing@pytorch.org>`__
```
[the remark this answers is not in the document]
Comment (svekars) [reply docs/source/community/contribution_guide.rst]: Changed it back for now. The forwarding was set up, but we tried to test it and it does not seem to work yet. Can update this later.

File: docs/source/community/contribution_guide.rst (hunk 3 above)
Comment (albanD) [reply docs/source/community/contribution_guide.rst]: Slack is not very restricted. Every application gets accepted.

**Review by albanD — APPROVED** [review]
Thanks for the updates! Looks good to me!

File: docs/source/community/contribution_guide.rst (hunk 3 above)
Comment (orionr) [reply docs/source/community/contribution_guide.rst]: Interesting - I was going by https://discuss.pytorch.org/t/request-to-join-slack-channel/16614 but maybe we should change that policy over time?

**Comment by clee2000** [comment]
@pytorcbot merge -g

**Pull request closed without merging**